frame/staking/src/lib.rs

```diff
@@ -1420,9 +1452,27 @@ decl_module! {
 		///  indirectly user-controlled. See [`unbond`] for more detail.
 		/// - Contains a limited number of reads, yet the size of which could be large based on `ledger`.
 		/// - Writes are limited to the `origin` account key.
+		/// ---------------
+		/// Complexity O(S) where S is the number of slashing spans to remove
+		/// Base Weight:
+		/// Update: 50.52 + .028 * S µs
+		/// - Reads: EraElectionStatus, Ledger, Current Era, Locks, [Origin Account]
+		/// - Writes: [Origin Account], Locks, Ledger
+		/// Kill: 79.41 + 2.366 * S µs
+		/// - Reads: EraElectionStatus, Ledger, Current Era, Bonded, Slashing Spans, [Origin Account], Locks
+		/// - Writes: Bonded, Slashing Spans (if S > 0), Ledger, Payee, Validators, Nominators, [Origin Account], Locks
+		/// - Writes Each: SpanSlash * S
 		/// # </weight>
-		#[weight = 400_000_000]
-		fn withdraw_unbonded(origin) {
+		#[weight = T::DbWeight::get().reads_writes(6, 6)
+			.saturating_add(80 * WEIGHT_PER_MICROS)
+			.saturating_add(
+				WEIGHT_PER_MICROS.saturating_mul(2).saturating_mul(Weight::from(*num_slashing_spans))
```

Comment:
I think the following syntax would have also worked:
```suggestion
				(2 * WEIGHT_PER_MICROS).saturating_mul(Weight::from(*num_slashing_spans))
```